swagger.yaml

```diff
@@ -1,42 +1,254 @@
+  /jobs:
+    get:
+      tags:
+        - "Public"
+      summary: Get a list of jobs.
+      description: Get a list of search reindex jobs that meet the criteria specified by the four query parameters.
+      parameters:
+        - $ref: '#/parameters/limit'
+        - $ref: '#/parameters/offset'
+        - $ref: '#/parameters/sort'
+        - $ref: '#/parameters/state'
+      produces:
+        - application/json
+      responses:
+        200:
+          description: Successfully returned the requested list of jobs.
+          schema:
+            $ref: "#/definitions/Jobs"
+        500:
+          $ref: '#/responses/InternalError'
+        400:
+          $ref: '#/responses/InvalidRequest'
+    post:
+      tags:
+        - "Public"
+      summary: Create a new search reindex job.
+      description: Post details of a new search reindex job and receive the new Job object created in response.
+      parameters:
+        - $ref: '#/parameters/job'
       produces:
         - application/json
       responses:
         200:
           description: OK
           schema:
-            $ref: "#/definitions/HelloResponse"
+            $ref: "#/definitions/Job"
         500:
           $ref: '#/responses/InternalError'
 
 responses:
+
   InternalError:
     description: "Failed to process the request due to an internal error"
 
+  ResourceNotFound:
+    description: "Failed to locate a Job that has the specified id"
+
+  InvalidRequest:
+    description: "The request does not contain the correct parameters and/or values in its parameters"
+
 definitions:
-  HelloResponse:
+
+  Job:
+    type: object
+    required:
+      - "id"
+      - "last_updated"
+      - "links"
+      - "search_index_name"
+      - "state"
+    properties:
+      id:
+        type: string
+      last_updated:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      links:
+        type: object
+        required:
+          - "tasks"
+          - "self"
+        properties:
+          tasks:
+            type: string
+            format: http://localhost:<port>/jobs/{id}/tasks
+          self:
+            type: string
+            format: http://localhost:<port>/jobs/{id}
+      number_of_tasks:
+        type: integer
+      reindex_completed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_failed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_started:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      search_index_name:
+        type: string
+      state:
+        type: string
+        enum:
+          - CREATED
+          - IN-PROGRESS
+          - COMPLETED
+          - FAILED
+      total_search_documents:
+        type: integer
+
+  NewJob:
     type: object
+    required:
+      - "last_updated"
+      - "search_index_name"
+      - "state"
     properties:
-      message:
+      last_updated:
```

Comment:
All fields containing ISO8601 timestamps should be of format `date-time`, nice to always include an example and explicit descriptions for each field.

Incl. `reindex_started`, `reindex-completed` and `reindex-failed`

Reply:
examples and descriptions now added but I'm not sure if the examples I've used are correct?

Reply:
I've found the following example of an ISO8601 timestamp so I've replaced my previous examples with this: 2021-03-18T09:03:41+0000